### PR TITLE
fix(third_party): fix UniversalReference class header page template

### LINKED ISSUE #345: Retrieve header page syntax
Currently on UniversalReference pages, syntax portions are left out of the `class.header` pages. They should be restored and match the ManagedReference pattern as well.

### PR DESCRIPTION
**Pull request by dandhlee**
Syntax definitions were left out of UniversalReference template for `class.header.tmpl.partial`. Adding them back in, so that the information is captured and presented in our documentation set. Also retrieving parameters and returns section where needed.

Fixes #345 and b/218725782. Towards b/208866895.

- [x] Tests pass

### CODE REVIEW

File: third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial [hunk 1]
```diff
@@ -1,12 +1,51 @@
...
+  <pre class="prettyprint{{#wrapdeclarationcode}} wrap-code{{/wrapdeclarationcode}}"><code>{{syntax.content.0.value}}</code></pre>
+</div>
+{{/syntax.content.0}}
 {{#summary}}
 <div class="markdown level0 summary">{{{summary}}}</div>
 {{/summary}}
 {{#conceptual}}
 <div class="markdown level0 conceptual">{{{conceptual}}}</div>
 {{/conceptual}}
+{{#syntax.parameters.0}}
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial:15]: I don't see any golden updates from this parameters and returns addition. Do we need it?

**Review by tbpg — APPROVED** [review]

**Comment by dandhlee** [comment]
Need to make few more adjustments before pushing this live. Will request for another review once I have few more updates through.

File: third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial (hunk 1 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial:15]: Updated the goldens to include ones with parameters. `bigquery_stroage` doesn't seem to have any docstrings on returns from constructors.

**Comment by dandhlee** [comment]
Merged the content of #352 and made minor edits. Updated the base test data and the goldens. Please take a look again!

**Review by tbpg — APPROVED** [review]

**Comment by dandhlee** [comment]
Waiting until https://github.com/googleapis/sphinx-docfx-yaml/pull/181 is merged and new version is released.

**Comment by dandhlee** [comment]
https://github.com/googleapis/sphinx-docfx-yaml/pull/180: new release has been successfully rolled out. Merging.

**Pull request merged**